Assessment/Assessment/ObjectTraits.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+namespace MemMgmt
+{
+
+template<typename T>
+class object_traits

```

Comment:
You can stick with some class naming convention everyware, ObjectTraits or object_traits.